quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,461 @@
+
+    // TODO - Config (Optional): Change the authentication method from User Prompt to any of the other options
+    //  Some of the auth modes require additional environment variables to be set in order to work (see usage in generateConnectionString below)
+    //  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+    private static final String AUTHENTICATION_MODE = "UserPrompt"; // Options: (UserPrompt|ManagedIdentity|AppKey|AppCertificate|DeviceCode)
+
+    // TODO - Config (Optional): Toggle to false to execute this script "unattended"
+    private static final boolean WAIT_FOR_USER = true;
+
+    private static String databaseName;
+    private static String tableName;
+    private static String kustoUrl;
+    private static String ingestUrl;
+    private static boolean useExistingTable;
+    private static boolean alterTable;
+    private static boolean queryData;
+    private static boolean ingestData;
+    private static List<Map<String, String>> dataToIngest;
+    private static String tableSchema;
+    private static int step = 1;
+    private static final String BATCHING_POLICY = "{ \"MaximumBatchingTimeSpan\": \"00:00:10\", \"MaximumNumberOfItems\": 500, \"MaximumRawDataSizeMB\": 1024 }";
+    private static final int WAIT_FOR_INGEST_SECONDS = 20;
+
+    public static void main(String[] args) throws InterruptedException, IOException {
+        System.out.println("Kusto sample app is starting...");
+
+        loadConfigs(KustoSampleApp.CONFIG_FILE_NAME);
+
+        if (AUTHENTICATION_MODE.equals("UserPrompt")) {
+            waitForUserToProceed("You may be prompted more than once for credentials during this script. Please return to the console after authenticating.");
+        }
+
+        try (IngestClient ingestClient = IngestClientFactory.createClient(generateConnectionString(ingestUrl, AUTHENTICATION_MODE))) {
+            // Tip: Avoid creating a new Kusto/ingest Client for each use. Instead, create the clients once and reuse them.
+            Client kustoClient = new ClientImpl(generateConnectionString(kustoUrl, AUTHENTICATION_MODE));
+
+            if (useExistingTable) {
+                if (alterTable) {
+                    // Tip: Usually table was originally created with a schema appropriate for the data being ingested, so this wouldn't be needed.
+                    // Learn More: For more information about altering table schemas, see: https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/alter-table-command
+                    alterExistingTableToProvidedSchema(kustoClient, databaseName, tableName, tableSchema);
+                }
+
+                if (queryData) {
+                    // Learn More: For more information about Kusto Query Language (KQL), see: https://docs.microsoft.com/azure/data-explorer/write-queries
+                    queryExistingNumberOfRows(kustoClient, databaseName, tableName);

```

Comment:
I don't think it makes sense to query existing rows if it's a new table.

On the other hand, a user may want "Query-only" mode. In that case, though, I suggest that between this and the 2nd set of queries, better to run the 2nd (since it paints a more complete picture with 2 queries instead of 1).